swift/Arrow/Sources/Arrow/ArrowCImporter.swift

```diff
@@ -0,0 +1,143 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import ArrowC
+
+public class ArrowCImporter {
+    private func appendToBuffer(
+        _ cBuffer: UnsafeRawPointer?,
+        arrowBuffers: inout [ArrowBuffer],
+        byteCount: Int,
+        length: UInt,
+        nullCount: UInt? = nil) {
+        if cBuffer == nil {
+            arrowBuffers.append(ArrowBuffer.createEmptyBuffer())
+            return
+        }
+
+        let allocator = MemoryAllocator(64)
+        let pointer = allocator.allocateArray(byteCount)
+        pointer.copyMemory(from: cBuffer!, byteCount: byteCount)
+        arrowBuffers.append(ArrowBuffer(length: length, capacity: UInt(byteCount), rawPointer: pointer))
+    }
+
+    public init() {}
+
+    public func importType(_ cArrow: String, name: String = "") ->
+        Result<ArrowField, ArrowError> {
+        do {
+            let type = try ArrowType.fromCDataFormatId(cArrow)
+            return .success(ArrowField(name, type: ArrowType(type.info), isNullable: true))
+        } catch {
+            return .failure(.invalid("\(error)"))
+        }
+    }
+
+    public func importField(_ cSchema: ArrowC.ArrowSchema) ->
+        Result<ArrowField, ArrowError> {
+        if cSchema.n_children > 0 {
+            return .failure(.invalid("Children currently not supported"))
+        } else if cSchema.dictionary != nil {
+            return .failure(.invalid("Dictinoary types currently not supported"))
+        }
+
+        switch importType(
+            String(cString: cSchema.format), name: String(cString: cSchema.name)) {
+        case .success(let field):
+            release(cSchema)
+            return .success(field)
+        case .failure(let err):
+            return .failure(err)
+        }
+    }
+
+    public func importArray(
+        _ cArray: ArrowC.ArrowArray,
+        arrowType: ArrowType,
+        isNullable: Bool = false) -> Result<ArrowArrayHolder, ArrowError> {
+        let arrowField = ArrowField("", type: arrowType, isNullable: isNullable)
+        return importArray(cArray, arrowField: arrowField)
+    }
+
+    public func importArray(
+        _ cArray: ArrowC.ArrowArray,
+        arrowField: ArrowField) -> Result<ArrowArrayHolder, ArrowError> {
+        if cArray.null_count < 0 {
+            return .failure(.invalid("Uncomputed null count is not supported"))
+        } else if cArray.n_children > 0 {
+            return .failure(.invalid("Children currently not supported"))
+        } else if cArray.dictionary != nil {
+            return .failure(.invalid("Dictinoary types currently not supported"))
+        }
+
+        let arrowType = arrowField.type
+        let length = UInt(cArray.length)
+        let nullCount = UInt(cArray.null_count)
```

Comment:
I'm curious: why `UInt`? It looks like this makes you convert back-and-forth between signed and unsigned.